### PR TITLE
docs: add numbering to headers to help navigation and readability

### PR DESCRIPTION
**Pull request by smccarthy-ie**
Mainly adding numbered headings to help keep track of where you are in procedures, and minor copy edits only. No changes to technical content.  

Please cherry pick to v0.7 also. Thanks

### CODE REVIEW

File: doc/user-guides/secure-protect-connect-single-multi-cluster.md [hunk 1]
```diff
@@ -1,43 +1,45 @@
-# Secure, protect, and connect APIs with Kuadrant
+# Secure, protect, and connect APIs with Kuadrant on OpenShift
```
[the remark this answers is not in the document]
Comment (maleck13) [reply doc/user-guides/secure-protect-connect-single-multi-cluster.md:1]: I will create a follow up for that https://github.com/Kuadrant/kuadrant-operator/issues/640

[the remark this answers is not in the document]
Comment (smccarthy-ie) [reply doc/user-guides/secure-protect-connect-single-multi-cluster.md:1]: Thanks @maleck13 Yes, agree that a single generic guide would be best and easier to maintain. 

We currently have two guides with the same heading name. This PR updates both heading names to distinguish between OpenShift and Kubernetes for now, which matches the approach taken in the site TOC. 

See https://docs.kuadrant.io/0.7.0/kuadrant-operator/doc/user-guides/secure-protect-connect-single-multi-cluster/

File: doc/user-guides/secure-protect-connect-single-multi-cluster.md [hunk 2]
```diff
@@ -408,26 +424,27 @@ export rootDomain=example.com
...
-Use `kuadrantctl` to generate your `HTTPRoute`.
+You will use `kuadrantctl` to generate your `HTTPRoute`.
 
 NOTE: The sample OAS has some placeholders for namespaces and domains. You will inject valid values into these placeholders based on your previous environment variables.
 
-Generate the resource from your OAS, (`envsubst` will replace the placeholders):
+Generate the resource from your OAS as follows, (`envsubst` will replace the placeholders):
 
 ```bash
 cat $oasPath | envsubst | kuadrantctl generate gatewayapi httproute --oas -
```
Comment (maleck13) [inline doc/user-guides/secure-protect-connect-single-multi-cluster.md:440]:
```suggestion
cat $oasPath | envsubst | kuadrantctl generate gatewayapi httproute --oas - | kubectl apply -f -
```

Comment (maleck13) [reply doc/user-guides/secure-protect-connect-single-multi-cluster.md:440]: spotted a  mistake in the command there

**Review by maleck13 — APPROVED** [review]
changes look good. Once merged will cherry pick to the 0.7 branch

**Comment by smccarthy-ie** [comment]
Thanks @maleck13 Can you merge this one also? I don't have write access to Merge button :)

**Pull request merged**